### DISCUSSION TITLE
Do people use mask on Alpine?

### DISCUSSION THREAD

**Discussion by 8Mobius8** [jacobdeichert/mask#92] [Ideas]
I am curious how many other might be using mask on Alpine.

I was trying to include this tool in some in a devcontainer that would support development across different repos, in a sort of tool chain way, by using Microsoft's alpine devcontainers.

In both use cases below I am installing `mask` on an alpine docker image:
`mcr.microsoft.com/vscode/devcontainers/base:0-alpine-3.15`

Sadly using `wget` downloading the linux distribution and installing doesn't really work:
```sh
$ /usr/local/bin/mask
bash: /usr/local/bin/mask: No such file or directory
$ file /usr/local/bin/mask
/usr/local/bin/mask: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=09866be28414d1e84474b259260649c7388016f3, for GNU/Linux 3.2.0, with debug_info, not stripped
```

Looks like its using a dynamic linker that isn't supported in Alpine. I guess `musl` is supported however.

Interestingly using cargo to install `mask` it picks up the `musl` platform just fine:
```sh
$ cargo install mask
    Updating crates.io index
  Downloaded mask v0.11.2
  Downloaded 1 crate (13.4 KB) in 1.18s
  Installing mask v0.11.2
  
   ...
   
   Compiling mask-parser v0.2.1
   Compiling mask v0.11.2
    Finished release [optimized] target(s) in 3m 36s
  Installing /home/vscode/.cargo/bin/mask
   Installed package `mask v0.11.2` (executable `mask`)
warning: be sure to add `/home/vscode/.cargo/bin` to your PATH to be able to run the installed binaries
$ /home/vscode/.cargo/bin/mask 
WARNING: no maskfile.md found
error: 'mask' requires a subcommand, but one was not provided

USAGE:
    mask [FLAGS] [OPTIONS]

For more information try --help
$ /home/vscode/.cargo/bin/mask --version
WARNING: no maskfile.md found
mask 0.11.2

$ file /home/vscode/.cargo/bin/mask 
/home/vscode/.cargo/bin/mask: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-x86_64.so.1, with debug_info, not stripped
```
You can see in the interpreter section of the musl is used instead of the linux.

Not sure what it would take to get another binary added on release for `musl` or if its worth the effort but it would save time for use cases where installing cargo on alpine distributions can take a long time.

**Comment by jacobdeichert** [comment]
> Not sure what it would take to get another binary added on release for musl or if its worth the effort but it would save time for use cases where installing cargo on alpine distributions can take a long time.

I think adding a musl binary to the release assets makes sense 👍 

Other rust tools on GitHub offer many types of precompiled linux binaries, so I assume its pretty common to generate a musl version as well.

Feel free to open an issue! I currently don't have time to work on this, but others might choose to.

**Comment by 8Mobius8** [comment]
> > Not sure what it would take to get another binary added on release for musl or if its worth the effort but it would save time for use cases where installing cargo on alpine distributions can take a long time.
> 
> I think adding a musl binary to the release assets makes sense 👍
> 
> Other rust tools on GitHub offer many types of precompiled linux binaries, so I assume its pretty common to generate a musl version as well.
> 
> Feel free to open an issue! I currently don't have time to work on this, but others might choose to.

Thanks for the go ahead! I create a small description of an issue #93
